### ISSUE TITLE
[Feature]: Developers should only be able to apply manifests to spaces they have access to

### ISSUE THREAD

**Issue by gcapizzi** [cloudfoundry/cf-k8s-controllers#741]
Overlap with #699 

### Background

**As an** admin
**I want** to assign the `SpaceDeveloper` role to the users I trust
**So that** they can push apps to their spaces

### Acceptance Criteria

**GIVEN** two spaces `space1` and `space2` exist
**AND** `app1` exists in `space1`
**AND** `app2` exists in `space2`
**AND** `mgr` has a `RoleBinding` to the `SpaceManager` `ClusterRole` in `space1`
**AND** `dev` has a `RoleBinding` to the `SpaceDeveloper` `ClusterRole` in `space1`
* **WHEN I** [apply the following manifest](https://v3-apidocs.cloudfoundry.org/version/3.116.0/index.html#apply-a-manifest-to-a-space) to `space2` as `dev`:
  ```
  version: 1
  applications:
  - name: app2
    env:
      FOO: bar
  ```
  **THEN I** get the following response:
    ```
    HTTP/1.1 404 Not Found
    Content-Type: application/json
    ```
    ```json
    {
      "errors": [
        {
          "code": 10010,
          "detail": "Space not found",
          "title": "CF-ResourceNotFound"
        }
      ]
    }
    ```
* **WHEN I** apply the following manifest to `space1` as `mgr`:
  ```
  version: 1
  applications:
  - name: app1
    env:
      FOO: bar
  ```
  **THEN I** get the following response:
    ```
    HTTP/1.1 403 Forbidden
    Content-Type: application/json
    ```
    ```json
    {
      "errors": [
        {
          "code": 10003,
          "detail": "You are not authorized to perform the requested action",
          "title": "CF-NotAuthorized"
        }
      ]
    }
    ```
* **WHEN I** apply the following manifest to `space1` as `mgr`:
  ```
  version: 1
  applications:
  - name: app1
     processes:
    - type: web
      instances: 3
  ```
  **THEN I** get the following response:
    ```
    HTTP/1.1 403 Forbidden
    Content-Type: application/json
    ```
    ```json
    {
      "errors": [
        {
          "code": 10003,
          "detail": "You are not authorized to perform the requested action",
          "title": "CF-NotAuthorized"
        }
      ]
    }
    ```

### Dev Notes

This story should introduce authentication to the following repository methods:

* `AppRepo.GetAppByNameAndSpace`
* `AppRepo.CreateOrPatchAppEnvVars`
* `ProcessRepo.CreateProcess`
* `RouteRepo.ListRoutesForApp`

These methods are the only ones not being used by other endpoints at the moment. While acceptance via user facing behaviour might not be possible, we still want to make sure that all these methods are authenticated and that the privileged client is used only when absolutely necessary.

**Comment by gcapizzi** [comment]
Rejecting this as including apps the user can't access in the manifest leads to an `App not found` error: 
```
$ cf login # dev
$ cf curl -i "/v3/spaces/[space2]/actions/apply_manifest" \
  -H "Content-type: application/x-yaml" \
  -d '---
version: 1
applications:
- name: app2
  env:
    FOO: bar'
HTTP/1.1 404 Not Found
...
{"errors":[{"detail":"App not found. Ensure it exists and you have access to it.","title":"CF-ResourceNotFound","code":10010}]}
```

The error is incorrect, as when the app doesn't actually exist we create it. If the user doesn't have access to the space, we should return `Space not found`.

**Comment by matt-royal** [comment]
I'm picking this up to add the missing error message. Thanks, Giuseppe.

**Comment by matt-royal** [comment]
I fixed the issue in PR #894
